tests/providers/cncf/kubernetes/utils/test_pod_manager.py

```diff
@@ -403,3 +421,77 @@ def test_container_is_running(remote_pod, result):
+            ([b"message1\n", b"message2"], [b"message1\n", b"message2"]),
+            ([b"first_part", b"_second_part"], [b"first_part_second_part"]),
+            ([b""], [b""]),
+        ],
+    )
+    def test_chunks(self, chunks, expected_logs):
+        with mock.patch.object(PodLogsConsumer, "logs_available") as logs_available:
+            logs_available.return_value = True
+            consumer = PodLogsConsumer(
+                response=mock.MagicMock(stream=mock.MagicMock(return_value=chunks)),
+                pod=mock.MagicMock(),
+                pod_manager=mock.MagicMock(container_is_running=mock.MagicMock(return_value=True)),
+                container_name="base",
+            )
+            assert list(consumer) == expected_logs
+
+    def test_container_is_not_running(self):
+        with mock.patch.object(PodLogsConsumer, "logs_available") as logs_available:
+            logs_available.return_value = False
+            consumer = PodLogsConsumer(
+                response=mock.MagicMock(stream=mock.MagicMock(return_value=[b"message1", b"message2"])),
+                pod=mock.MagicMock(),
+                pod_manager=mock.MagicMock(container_is_running=mock.MagicMock(return_value=False)),
+                container_name="base",
+            )
+            assert list(consumer) == []
+
+    @pytest.mark.parametrize(
+        "container_run, termination_time, now_time, timeout, expected_logs_available",
+        [
+            (False, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 1, 0, 0), 120, True),
+            (False, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 2, 0, 0), 120, False),
+            (False, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 5, 0, 0), 120, False),
+            (True, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 1, 0, 0), 120, True),
+            (True, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 2, 0, 0), 120, True),
+            (True, datetime(2022, 1, 1, 0, 0, 0, 0), datetime(2022, 1, 1, 0, 5, 0, 0), 120, True),
+        ],
+    )
+    @mock.patch("airflow.providers.cncf.kubernetes.utils.pod_manager.container_is_running")
+    @mock.patch("airflow.providers.cncf.kubernetes.utils.pod_manager.get_container_status")
+    def test_logs_available(
+        self,
+        mock_get_container_status,
+        mock_container_is_running,
+        container_run,
+        termination_time,
+        now_time,
+        timeout,
+        expected_logs_available,
+    ):
+        mock_container_is_running.return_value = container_run
+        mock_get_container_status.return_value = mock.MagicMock(
+            state=mock.MagicMock(terminated=mock.MagicMock(finished_at=termination_time))
+        )
+        with mock.patch(
+            "airflow.providers.cncf.kubernetes.utils.pod_manager.datetime",
+            mock.MagicMock(now=mock.MagicMock(return_value=now_time), fromtimestamp=datetime.fromtimestamp),
+        ):
```

Comment:
In general we use [`time-machine`](https://github.com/adamchainz/time-machine), previously `freezegun`, for mock everything related to current datetime rather than manually mock it.

For sure there are some tests exists which use mock for datetime/time however better to use time-machine (as context manager or fixture).

In additional I'm not familiar with k8s objects, and do not know is [`V1ContainerStateTerminated.finished_at`](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ContainerStateTerminated.md) return timezone-naive or timezone-aware datetime.

Reply:
Thank you for pointing out the [time-machine](https://github.com/adamchainz/time-machine) - it solves lots of problems.
Speaking about the [V1ContainerStateTerminated.finished_at](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ContainerStateTerminated.md), experiments show that it retrieves datetime in UTC. I also added timezone specifications to the unit test.